templates/aws-stack.yml

```diff
@@ -400,6 +404,28 @@ Parameters:
       - "false"
     Default: "false"
 
+  StartSchedule:
+    Type: String
+    Description: "The recurring schedule for changing agent count, in Unix cron syntax format."
+    Default: ""
+    AllowedPattern: "^$|^(((\\d+,)+\\d+|(\\d+(\\/|-)\\d+)|\\d+|\\*) ?){5,7}$"
```

Comment:
Any citation for this regexp? Or have you written it just for this? I don't see anything wrong with it, but regular expressions can be a bit “write only” :) And unfortunately we can't really unit test it here. Anyway, pretty low impact if it does/doesn't allow something valid, we can iterate on it.

For reference — the [`AWS::AutoScaling::ScheduledAction` documentation for `Recurrence`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-as-scheduledaction.html#aws-resource-as-scheduledaction-properties) says:

> The recurring schedule for this action, in Unix cron syntax format. For more information about cron syntax, see [Crontab](http://crontab.org).

Reply:
Written by myself unfortunately (for both of us). Mainly added in as the error you get here if it's wrong is pretty obscure, I think the only way I found out what was wrong was by tracing the error message back to the library that AWS is using for the scheduling.

I've tested manually with https://regexr.com/ a fair bit, but happy to remove this check if you want to err on the side of not accidentally blocking valid expressions.